pkg/controller/apicuriodeployment/apicuriodeployment_controller.go

```diff
@@ -0,0 +1,257 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"fmt"
+	openshift "github.com/integr8ly/apicurio-operator/pkg/apis/integreatly/openshift/client"
+	v1template "github.com/openshift/api/template/v1"
+	kuberr "k8s.io/apimachinery/pkg/api/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+	"github.com/openshift/api/image/v1"
+	v12 "github.com/openshift/api/apps/v1"
+)
+
+// Add creates a new ApicurioDeployment Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileApiCurioDeployment{
+		client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+		config: mgr.GetConfig(),
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	opts := controller.Options{Reconciler: r}
+	c, err := controller.New("apicuriodeployment-controller", mgr, opts)
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource ApicurioDeployment
+	err = c.Watch(&source.Kind{Type: &integreatlyv1alpha1.ApicurioDeployment{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner ApicurioDeployment
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &integreatlyv1alpha1.ApicurioDeployment{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileApiCurioDeployment{}
+
+
+
+func (r *ReconcileApiCurioDeployment) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("Reconciling ApicurioDeployment %s/%s\n", request.Namespace, request.Name)
+
+	// Fetch the ApicurioDeployment instance
+	instance := &integreatlyv1alpha1.ApicurioDeployment{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+
+		return reconcile.Result{}, err
+	}
+
+
+	err = r.bootstrap(request)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	exts, err := r.processTemplate(instance)
+	if err != nil {
+		logrus.Errorf("Error while processing template: %v", err)
```

Comment:
The code generated by the operator-sdk uses `log.Printf` instead of `logrus` (see line 79). I'd be in favor of using that same built in logging. Or is there a big advantage to using logrus?

Reply:
I think it used logrus in the past version so it could be a bad habit on my part :)